### ISSUE TITLE
[BUG] Too many map executions shown in Flyte console

### ISSUE THREAD

**Issue by fg91** [flyteorg/flyte#4531]
### Describe the bug

If one creates a map task over e.g. 5 inputs and on the platform side configures e.g. 2 log links for map tasks, FlyteConsole shows 10 map executions instead of 5.

### Expected behavior

The number of executions shown in FlyteConsole should be equal to the number of pods.

### Additional context to reproduce

[Example](https://docs.flyte.org/projects/cookbook/en/latest/auto_examples/advanced_composition/map_task.html#map-a-task-with-multiple-inputs) from the docs:

```py
import functools
from flytekit import map_task, task, workflow


@task
def multi_input_task(quantity: int, price: float, shipping: float) -> float:
    return quantity * price * shipping


@workflow
def multiple_workflow(list_q: list[int] = [1, 2, 3, 4, 5], p: float = 6.0, s: float = 7.0) -> list[float]:
    partial_task = functools.partial(multi_input_task, price=p, shipping=s)
    return map_task(partial_task)(quantity=list_q)
```

### Screenshots

For two log links configured, FlyteConsole shows 10 executions instead of 5:

![image](https://github.com/flyteorg/flyte/assets/36511035/3296bb72-8ce4-401b-8b48-d106859754c3)

If we configure a third log link, the number of shown executions is 15.

### Are you sure this issue hasn't been raised already?

- [X] Yes

### Have you read the Code of Conduct?

- [X] Yes

**Comment by fg91** [comment]
> @fg91 , do you also see the same with array node map tasks?

Yes, we do 👍

**Comment by fg91** [comment]
The Flyteconsole/UI revamp released with 1.12 appears to have fixed this.
